File: sdk/ios/SsdidCore/Sources/SsdidCore/Api/IdentityApi.swift
import Foundation

/// Facade for identity lifecycle operations.
public struct IdentityApi {
    let vault: Vault
    let client: SsdidClient

    public func create(name: String, algorithm: Algorithm) async throws -> Identity {
        try await client.initIdentity(name: name, algorithm: algorithm)
    }

    public func list() async -> [Identity] {
        await vault.listIdentities()
    }

    public func get(keyId: String) async -> Identity? {
        await vault.getIdentity(keyId: keyId)
    }

    public func delete(keyId: String) async throws {
        try await vault.deleteIdentity(keyId: keyId)
    }

    public func buildDidDocument(keyId: String) async throws -> DidDocument {
        try await vault.buildDidDocument(keyId: keyId)
    }

    public func updateDidDocument(keyId: String) async throws {
        try await client.updateDidDocument(keyId: keyId)
    }

    public func deactivate(keyId: String) async throws {
        try await client.deactivateDid(keyId: keyId)
    }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/api/IdentityApi.kt
package my.ssdid.sdk.api

import my.ssdid.sdk.domain.SsdidClient
import my.ssdid.sdk.domain.model.*
import my.ssdid.sdk.domain.vault.Vault

class IdentityApi internal constructor(
    private val vault: Vault,
    private val client: SsdidClient
) {
    suspend fun create(name: String, algorithm: Algorithm): Result<Identity> =
        client.initIdentity(name, algorithm)
    suspend fun list(): List<Identity> = vault.listIdentities()
    suspend fun get(keyId: String): Identity? = vault.getIdentity(keyId)
    suspend fun delete(keyId: String): Result<Unit> = vault.deleteIdentity(keyId)
    suspend fun buildDidDocument(keyId: String): Result<DidDocument> = vault.buildDidDocument(keyId)
    suspend fun updateDidDocument(keyId: String): Result<Unit> = client.updateDidDocument(keyId)
    suspend fun deactivate(keyId: String): Result<Unit> = client.deactivateDid(keyId)
}
